pkg/maps/configmap/configmap.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2016-2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configmap
+
+import (
+	"bytes"
+	"fmt"
+	"unsafe"
+
+	"github.com/cilium/cilium/pkg/bpf"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+)
+
+var log = logging.DefaultLogger.WithField(logfields.LogSubsys, "map-config")
+
+const (
+	// MapNamePrefix is the basename prefix of endpoint config maps.
+	MapNamePrefix = "cilium_ep_config_"
+
+	// MaxEntries represents the maximum number of elements in the map
+	MaxEntries = 1
+
+	// SkipPolicyIngress causes ingress policy to be skipped.
+	SkipPolicyIngress = 1 << 0
+	// SkipPolicyEgress causes ingress policy to be skipped.
+	SkipPolicyEgress = 1 << 1
+)
+
+var (
+	flagsToString = map[int]string{
+		SkipPolicyIngress: "SKIP_POLICY_INGRESS",
+		SkipPolicyEgress:  "SKIP_POLICY_EGRESS",
+	}
+
+	binMap = map[uint]string{
+		0: "POLICY_INGRESS",
+		1: "POLICY_EGRESS",
+	}
+)
+
+// Flags is a set of endpoint configuration flags interpreted by BPF code.
+//
+// Must be in sync with the enum ep_cfg_flag in <bpf/lib/eps.h>
+type Flags uint32
+
+// String converts the specified flags into a human-readable form.
+func (f Flags) String() string {
+	var buffer bytes.Buffer
+	for i := uint(0); i < 32; i++ {
+		bitVal := int(f & (1 << i))
+		if bitVal > 0 {
+			if flag, ok := flagsToString[bitVal]; ok {
+				buffer.WriteString(flag)
+				buffer.WriteString(",")
+			}
```

Comment:
Not a biggie, but `} else { buffer.WriteString("%04x", 1 << i) }` ?

Ie, if it can't be found in the map to convert it to a string, then print the raw bit value.

This would mean that if we somehow upgrade/downgrade and don't change the map (so it could have bit values not understood by this userspace), then it would at least print that the underlying map contains a non-zero bitvalue for this offset.